[processor/redaction] Introduce `blocked_key_patterns` parameter (#37664)
open-telemetry/opentelemetry-collector-contrib: processor/redactionprocessor/README.md

    # Any keys in this list are allowed so they don't need to be in both lists.
    ignored_keys:
      - safe_attribute

>     # blocked_key_patterns is a list of blocked span attribute key patters. Span attributes

Reviewer: ```suggestion
    # blocked_key_patterns is a list of blocked span attribute key patterns. Span attributes
```

If the value matches the regular expression for a blocked value only, the matching
part of the value is masked with a fixed length of asterisks.

`blocked_key_patterns` applies to the values of the keys matching one of the patterns.

> The value is then masked with a fixed length of asterisks or hashed by a function
> defined in `hash_function`.

Reviewer: nit (to make it a bit more future-proof)
```suggestion
The value is then masked according to the configuration.
```
